File: output/gpt4/js/Solution05_gpt.js
/**
 * @param {number} n
 * @return {number}
 */
var climbStairs = function(n) {
    if (n === 1 || n === 2) {
        return n;
    }
    
    let prevPrev = 1;
    let prev = 2;
    let current;
    
    for (let i = 3; i <= n; i++) {
        current = prevPrev + prev;
        prevPrev = prev;
        prev = current;
    }
    
    return current;
};

File: output/gpt4/java/Solution05_gpt.java
public class Solution05_gpt {
    public int climbStairs(int n) {
        if (n == 1 || n == 2) {
            return n;
        }
        
        int prevPrev = 1;
        int prev = 2;
        int current = 0;
        
        for (int i = 3; i <= n; i++) {
            current = prevPrev + prev;
            prevPrev = prev;
            prev = current;
        }
        
        return current;
    }
}
